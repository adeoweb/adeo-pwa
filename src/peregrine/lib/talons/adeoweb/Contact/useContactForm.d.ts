import * as React from 'react';
import { FormikErrors, FormikTouched } from 'formik';

type TUseContactFormProps<Values> = {
    initialValues?: Values;
};

export type TUseContactFormValues = {
    name: string;
    email: string;
    phone?: string;
    message: string;
};

type TUseContactForm<Values> = {
    handleSubmit: (e?: React.FormEvent<HTMLFormElement> | undefined) => void;
    handleChange: (
        eventOrPath: string | React.ChangeEvent<any>
    ) => void | ((eventOrTextValue: string | React.ChangeEvent<any>) => void);
    values: Values;
    errors: FormikErrors<Values>;
    touched: FormikTouched<Values>;
};

export function useContactForm<Values>(
    props: TUseContactFormProps<Values>
): TUseContactForm<Values>;
