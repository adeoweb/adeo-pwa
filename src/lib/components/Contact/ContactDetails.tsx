import React, { FunctionComponent } from 'react';
import { Col } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';

import classes from './styles/ContactDetails.scss';

const ContactDetails: FunctionComponent = () => {
    const { t } = useTranslation('common');

    return (
        <Col md={4}>
            <h2 className="light-title">
                <strong>{t('Contact Details')}</strong>
            </h2>
            <section className={classes.contactInfo}>
                <article>
                    <a className={classes.contactLine} href="tel:+37067647849">
                        +370 676 47 849
                    </a>
                    <a className={classes.contactLine} href="tel:+13122248950">
                        +1.312.224.8950
                    </a>
                    <a className={classes.contactLine} href="tel:+460709946900">
                        +46 (0) 709 946900
                    </a>
                </article>
                <article>
                    <a
                        className={classes.contactLine}
                        href="mailto:info@adeoweb.biz"
                    >
                        info@adeoweb.biz
                    </a>
                    <a
                        className={classes.contactLine}
                        href="mailto:sales@adeoweb.biz"
                    >
                        sales@adeoweb.biz
                    </a>
                    <a
                        className={classes.contactLine}
                        href="mailto:info@adeoweb.se"
                    >
                        info@adeoweb.se
                    </a>
                </article>
            </section>
        </Col>
    );
};

export default ContactDetails;
