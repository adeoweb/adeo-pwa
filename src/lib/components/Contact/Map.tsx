import React, { FunctionComponent } from 'react';
import classes from './styles/Map.scss';

interface MapProps {
    src: string;
    width: string;
    height: string;
    allowFullScreen?: boolean;
}

const defaultProps = {
    allowFullScreen: true
};

const Map: FunctionComponent<MapProps> = ({
    src,
    width,
    height,
    allowFullScreen = defaultProps.allowFullScreen
}) => {
    return (
        <div>
            <iframe
                className={classes.map}
                src={src}
                width={width}
                height={height}
                allowFullScreen={allowFullScreen}
                aria-hidden={'false'}
            />
        </div>
    );
};

export default Map;
