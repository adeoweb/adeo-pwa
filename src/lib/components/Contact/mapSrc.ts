export const mapSrc =
    'https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2293.6635289715405!2d23.928058015897946!3d54.908827680336074!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46e7188ef2c43c73%3A0xb87004cff71657cb!2sAdeo%20Web%20UAB!5e0!3m2!1sen!2slt!4v1590131925117!5m2!1sen!2slt';
