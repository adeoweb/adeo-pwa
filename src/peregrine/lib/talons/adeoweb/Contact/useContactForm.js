import { useFormik } from 'formik';
import * as yup from 'yup';

export const useContactForm = ({ initialValues = {} }) => {
    const handleFormSubmit = values => {
        console.log(values);
    };

    const formSchema = yup.object({
        name: yup.string().required(),
        email: yup.string().required().email(),
        phone: yup.string(),
        message: yup.string().required()
    });

    const { handleSubmit, handleChange, values, errors, touched } = useFormik({
        validationSchema: formSchema,
        initialValues,
        onSubmit: handleFormSubmit
    });

    return {
        handleSubmit,
        handleChange,
        values,
        errors,
        touched
    };
};
