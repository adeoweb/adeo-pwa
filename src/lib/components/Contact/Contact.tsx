import React, { FunctionComponent } from 'react';
import { Container, Row } from 'react-bootstrap';

import ContactDetails from './ContactDetails';
import ContactForm from './ContactForm';
import Map from './Map';
import { mapSrc } from './mapSrc';

const Contact: FunctionComponent = () => {
    return (
        <div className="page-main">
            <div className="page-title-wrapper">
                <Container>
                    <Map
                        src={mapSrc}
                        width={'100%'}
                        height={'460px'}
                        allowFullScreen={true}
                    />
                    <div className="mb-4" />
                    <Row>
                        <ContactForm />
                        <ContactDetails />
                    </Row>
                </Container>
            </div>
        </div>
    );
};

export default Contact;
