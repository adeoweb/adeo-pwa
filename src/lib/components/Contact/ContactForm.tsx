import React, { FunctionComponent } from 'react';
import { Form, Button, Col } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';

import { useContactForm } from 'src/peregrine/lib/talons/adeoweb/Contact/useContactForm';

const ContactForm: FunctionComponent = () => {
    const { t } = useTranslation(['common', 'customer']);

    const { handleSubmit, handleChange, values, errors, touched } =
        useContactForm();

    return (
        <Col md={8}>
            <h2 className="light-title">
                <strong>{t('common:Contact Us')}</strong>
            </h2>
            <Form noValidate onSubmit={handleSubmit}>
                <Form.Group className="required-field">
                    <Form.Label>{t('customer:Name')}</Form.Label>
                    <Form.Control
                        type="text"
                        name="name"
                        value={values.name}
                        onChange={handleChange}
                        isInvalid={!!(errors.name && touched.name)}
                        required
                    />
                    <Form.Control.Feedback type="invalid">
                        {errors.name}
                    </Form.Control.Feedback>
                </Form.Group>
                <Form.Group className="required-field">
                    <Form.Label>{t('customer:Email')}</Form.Label>
                    <Form.Control
                        type="email"
                        name="email"
                        value={values.email}
                        onChange={handleChange}
                        isInvalid={!!(errors.email && touched.email)}
                        required
                    />
                    <Form.Control.Feedback type="invalid">
                        {errors.email}
                    </Form.Control.Feedback>
                </Form.Group>
                <Form.Group>
                    <Form.Label>{t('customer:Phone Number')}</Form.Label>
                    <Form.Control
                        type="text"
                        name="phone"
                        value={values.phone}
                        onChange={handleChange}
                        isInvalid={!!(errors.phone && touched.phone)}
                    />
                    <Form.Control.Feedback type="invalid">
                        {errors.phone}
                    </Form.Control.Feedback>
                </Form.Group>
                <Form.Group className="required-field">
                    <Form.Label>
                        {t('customer:Whatâ€™s on your mind?')}
                    </Form.Label>
                    <Form.Control
                        as="textarea"
                        rows={1}
                        cols={30}
                        name="message"
                        value={values.message}
                        onChange={handleChange}
                        isInvalid={!!(errors.message && touched.message)}
                        required
                    />
                    <Form.Control.Feedback type="invalid">
                        {errors.message}
                    </Form.Control.Feedback>
                </Form.Group>
                <Button variant="primary" type="submit">
                    {t('common:Submit')}
                </Button>
            </Form>
        </Col>
    );
};

export default ContactForm;
