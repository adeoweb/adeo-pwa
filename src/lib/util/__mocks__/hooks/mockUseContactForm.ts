import { TUseContactForm } from 'src/peregrine/lib/talons/adeoweb/Contact/useContactForm';
import { FormikValues } from 'formik';

const mock: TUseContactForm<FormikValues> = {
    handleSubmit: () => {},
    handleChange: () => {},
    values: {},
    errors: {},
    touched: {}
};

export default mock;
