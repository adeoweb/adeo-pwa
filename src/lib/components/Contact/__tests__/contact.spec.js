import React from 'react';
import testRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Contact from 'src/lib/components/Contact';
import mockUseContactForm from 'src/lib/util/__mocks__/hooks/mockUseContactForm';

jest.mock('src/peregrine/lib/talons/adeoweb/Contact/useContactForm', () => {
    return {
        useContactForm: () => mockUseContactForm
    };
});

test('Contact page renders correctly', () => {
    const component = testRenderer.create(
        <Router>
            <Contact />
        </Router>
    );
    expect(component.toJSON()).toMatchSnapshot();
});
