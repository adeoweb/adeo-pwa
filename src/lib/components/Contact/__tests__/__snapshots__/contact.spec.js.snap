// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact page renders correctly 1`] = `
<div
  className="page-main"
>
  <div
    className="page-title-wrapper"
  >
    <div
      className="container"
    >
      <div>
        <iframe
          allowFullScreen={true}
          aria-hidden="false"
          className="map"
          height="460px"
          src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2293.6635289715405!2d23.928058015897946!3d54.908827680336074!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x46e7188ef2c43c73%3A0xb87004cff71657cb!2sAdeo%20Web%20UAB!5e0!3m2!1sen!2slt!4v1590131925117!5m2!1sen!2slt"
          title="contact-map"
          width="100%"
        />
      </div>
      <div
        className="mb-4"
      />
      <div
        className="row"
      >
        <div
          className="col-md-8"
        >
          <h2
            className="light-title"
          >
            <strong>
              common:Contact Us
            </strong>
          </h2>
          <form
            className=""
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="required-field form-group"
            >
              <label
                className="form-label"
              >
                customer:Name
              </label>
              <input
                className="form-control"
                name="name"
                onChange={[Function]}
                required={true}
                type="text"
                value="mock-name"
              />
              <div
                className="invalid-feedback"
              />
            </div>
            <div
              className="required-field form-group"
            >
              <label
                className="form-label"
              >
                customer:Email
              </label>
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                required={true}
                type="email"
                value="mock-email"
              />
              <div
                className="invalid-feedback"
              />
            </div>
            <div
              className="form-group"
            >
              <label
                className="form-label"
              >
                customer:Phone Number
              </label>
              <input
                className="form-control"
                name="phone"
                onChange={[Function]}
                type="text"
                value="mock-phone"
              />
              <div
                className="invalid-feedback"
              />
            </div>
            <div
              className="required-field form-group"
            >
              <label
                className="form-label"
              >
                customer:Whatâ€™s on your mind?
              </label>
              <textarea
                className="form-control"
                cols={30}
                name="message"
                onChange={[Function]}
                required={true}
                rows={1}
                value="mock-message"
              />
              <div
                className="invalid-feedback"
              />
            </div>
            <button
              className="btn btn-primary"
              disabled={false}
              type="submit"
            >
              common:Submit
            </button>
          </form>
        </div>
        <div
          className="col-md-4"
        >
          <h2
            className="light-title"
          >
            <strong>
              Contact Details
            </strong>
          </h2>
          <section
            className="contactInfo"
          >
            <article>
              <a
                className="contactLine"
                href="tel:+37067647849"
              >
                +370 676 47 849
              </a>
              <a
                className="contactLine"
                href="tel:+13122248950"
              >
                +1.312.224.8950
              </a>
              <a
                className="contactLine"
                href="tel:+460709946900"
              >
                +46 (0) 709 946900
              </a>
            </article>
            <article>
              <a
                className="contactLine"
                href="mailto:info@adeoweb.biz"
              >
                info@adeoweb.biz
              </a>
              <a
                className="contactLine"
                href="mailto:sales@adeoweb.biz"
              >
                sales@adeoweb.biz
              </a>
              <a
                className="contactLine"
                href="mailto:info@adeoweb.se"
              >
                info@adeoweb.se
              </a>
            </article>
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
